src/components/apply/Apply.js

```diff
@@ -1,11 +1,87 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { getFormValues, initialize, touch, blur } from 'redux-form';
+import _ from 'lodash';
+import { Tabs, Tab } from 'material-ui/Tabs';
 
 import ApplicantForm from 'src/components/apply/ApplicantForm';
+import { addJointApplicant, selectJointApplicant } from 'src/actions';
 
-const Apply = () => (
-  <div>
-    <ApplicantForm />
-  </div>
-);
+const tabStyle = {
+  fontSize: '11pt',
+  paddingRight: 5,
+  paddingLeft: 5,
+  fontWeight: 400,
+};
 
-export default Apply;
+/**
+ * Helper function that initializes the form with the new values provided and executes the validators/normalizers
+ * for each untouched field as soon as its loaded, causing error messages to be displayed immediately.
+ */
+const initializeApplyForm = (formData, touchActionDispatcher, blurActionDispatcher) => {
+  // Actually insert the new values into the form
+  initialize('apply', formData, false);
+
+  // for each field that has values, simulate a `BLUR` event to trigger validators and normalizers
+  _.each(formData, (fieldValue, fieldName) => {
+    touchActionDispatcher('apply', fieldName);
+    blurActionDispatcher('apply', fieldName, fieldValue);
+  });
+};
+
+const Apply = ({
+  curValues, jointApplicantData, selectedApplicantId, jointApplicantCount,
+  addJointApplicant, selectJointApplicant, initialize, touch, blur
+}) => {
+  const applicantTabs = _.map(_.range(jointApplicantCount), i => {
+    return <Tab style={tabStyle} label={`Joint Applicant ${i + 1}`} key={i} value={i} />;
+  });
+  // Add an additional tab at the end to add additional joint applicants
+  const allTabs = [
+    ...applicantTabs,
+    <Tab style={tabStyle} label={'+ Add Joint Applicant'} key={applicantTabs.length} value={applicantTabs.length} />
+  ];
+
+  const numberOfApplicantsAllowed = 4;
+  const handleTabClick = index => {
+    if(index === jointApplicantCount) {
```

Comment:
`addJointApplicant()` and `selectJointApplicant()` are bound redux action creators, meaning that they'll end up dispatching actions of their own.  Unless I'm really misunderstanding, they won't be accessible from within the reducer and even if they were, using them there would be a side effect and go against the principles of reducers.

Is there something I'm missing here?